### ISSUE TITLE
Bug

### ISSUE THREAD

**Issue by CageFox** [marijnkampf/Random-Photo-Screensaver#45]
RPS often falls with this exception:

Сигнатура проблемы:
CLR20r3
01:	RPS4.exe
02:	4.5.9.0
03:	55b67214
04:	RPS4
05:	4.5.9.0
06:	55b67214
07:	14b
08:	6e
09:	System.NullReferenceException
Version:	6.3.9600.2.0.0.256.48
Lang1049
1:	5809
2:	580900b529c7dbbc59a95eaef54aadb0
3:	6837
4:	6837a3a0b5aaa4c8bda65825368f8a17

**Comment by marijnkampf** [comment]
Are there any particular circumstances that this occurs?

**Comment by CageFox** [comment]
No, just randomly. It shows raw blue screen instead of photo and when I try to see my desktop back the error window appears

**Comment by marijnkampf** [comment]
Which version of Windows are you running?

**Comment by CageFox** [comment]
Windows 8 64bit

**Comment by CageFox** [comment]
Is there any way to find which of my thousands pictures causes the error? May be some debug log "showing picture .1.. ok\nshowing picture ..2. ok"
?

**Comment by marijnkampf** [comment]
Yes there is:
* In the RPS Configuration screen (Press S whilst screensaver is running or Right click desktop > Personalize > Screen Saver > Settings...)
* Select Slideshow tab
* Show advanced settings...
* Check Debug log

If you identify particular images / files that cause a crash I would really appreciate it if you would be able to post (a selection of) them to GitHub or email ( see http://www.abscreensavers.com/contact ), so I can create a fix for RPS.

**Comment by CageFox** [comment]
Ok. And where is this debug log location?

**Comment by CageFox** [comment]
BTW. This exception unhandled, but you can handle it and display informative message on the screen ;)

**Comment by marijnkampf** [comment]
Next to the Check Debug log checkbox there is a button "Open log location in Explorer". Typically this is C:\users\[user name]\AppData\Local\Random Photo Screensaver\debug_#data#.txt.

BTW That is why I'm hoping you can send me a sample file so I can fix the issue :)

**Comment by CageFox** [comment]
Ok. Thou the sample picture may contain very explicit content ;)

**Comment by marijnkampf** [comment]
Please keep it PG rated on GitHub, if the explicit content are the only ones causing a crash please use email address from contact page :/

**Comment by CageFox** [comment]
The trouble image not appears in the debug log. Last image in the log is ok, and then screensaver falling in blue screen without record in log
So, I still unable to detect one

**Comment by marijnkampf** [comment]
The debug log is written before the file is accessed so it should have written it before it crashed. It could be that the issue isn't a particular file as I expected. Would you either be able to run RPS on a subset of images and keep removing images until it crashes or navigate through your images in RPS using the shortcuts and seeing whether it crashes on a particular image?

**Comment by CageFox** [comment]
No, this not possible. I've got about 8000 pictures, it will take too long time to find buggy one. May be you can add some debug code and may be send me debug version with extended trace of exception stack?

**Comment by marijnkampf** [comment]
Fair enough, I've opened a separate issue to for adding a global exception catch https://github.com/marijnkampf/Random-Photo-Screensaver/issues/49

I'll let you know when it's available.
